### PR TITLE
NPI-3655 Improve readability of getVelPoly() function

### PR DESCRIPTION
**Pull request by treefern**
No functional changes, but a bunch of renamed variables to hopefully make it clearer what this function is doing.

### CODE REVIEW

File: gnssanalysis/gn_io/sp3.py [hunk 1]
```diff
@@ -477,28 +477,28 @@ def getVelPoly(sp3Df: _pd.DataFrame, deg: int = 35) -> _pd.DataFrame:
...
     est = sp3Df.unstack(1).EST[["X", "Y", "Z"]]
-    x = est.index.get_level_values("J2000").values
-    # x = est.index.values
-    y = est.values
+    time_values = est.index.get_level_values("J2000").values
+    xyz_positions = est.values
 
-    off, scl = mapparm([x.min(), x.max()], [-1, 1])  # map from input scale to [-1,1]
+    # map from input scale to [-1,1]
+    offset, scale_factor = mapparm([time_values.min(), time_values.max()], [-1, 1])
```
Comment (seballgeyer) [inline gnssanalysis/gn_io/sp3.py:484]: not sure we need `factor` scale can be enough

**Review by seballgeyer — COMMENTED** [review]
It is a good idea to make it clearer. 
However sometimes long variable name can make the code more difficult to read. And here some a quite long.

I did a few suggestions, I didn't went through all changes.  

Otherwise it is OK

File: gnssanalysis/gn_io/sp3.py [hunk 2]
```diff
@@ -477,28 +477,28 @@ def getVelPoly(sp3Df: _pd.DataFrame, deg: int = 35) -> _pd.DataFrame:
 
     """
     est = sp3Df.unstack(1).EST[["X", "Y", "Z"]]
-    x = est.index.get_level_values("J2000").values
-    # x = est.index.values
-    y = est.values
+    time_values = est.index.get_level_values("J2000").values
+    xyz_positions = est.values
```
[the remark this answers is not in the document]
Comment (treefern) [reply gnssanalysis/gn_io/sp3.py:481]: 👍 `positions` would be fine yeah. My original concern was that the variables being `x` and `y`, could've sounded like the x and y _components_ of positions.

File: gnssanalysis/gn_io/sp3.py [hunk 3]
```diff
@@ -477,28 +477,28 @@ def getVelPoly(sp3Df: _pd.DataFrame, deg: int = 35) -> _pd.DataFrame:
 
     """
     est = sp3Df.unstack(1).EST[["X", "Y", "Z"]]
-    x = est.index.get_level_values("J2000").values
-    # x = est.index.values
-    y = est.values
+    time_values = est.index.get_level_values("J2000").values
```
[the remark this answers is not in the document]
Comment (treefern) [reply gnssanalysis/gn_io/sp3.py:480]: `times` I guess; I just wanted to make it clearer that it's a series.
Will update 👍

File: gnssanalysis/gn_io/sp3.py (hunk 1 above)
Comment (treefern) [reply gnssanalysis/gn_io/sp3.py:484]: Just noticed it was `scale_factor` in some other similar code, which got introduced during a PR review:

https://github.com/GeoscienceAustralia/gnssanalysis/commit/fe81af2bef28727413acccf89b8c43a778d33aea#diff-330d45d4cfeafdbd9dcc218de027fe2763d140a22cf9763c6cf944f9ad0f1543R108

File: gnssanalysis/gn_io/sp3.py [hunk 4]
```diff
@@ -477,28 +477,28 @@ def getVelPoly(sp3Df: _pd.DataFrame, deg: int = 35) -> _pd.DataFrame:
...
 
-    x_new = off + scl * (x)
-    coeff = _np.polyfit(x=x_new, y=y, deg=deg)
+    normalised_times = offset + scale_factor * (times)
+    coeff = _np.polyfit(x=normalised_times, y=positions, deg=deg)
 
-    x_prev = off + scl * (x - 1)
-    x_next = off + scl * (x + 1)
+    time_prev = offset + scale_factor * (times - 1)
+    time_next = offset + scale_factor * (times + 1)
```
Comment (treefern) [inline gnssanalysis/gn_io/sp3.py:490]: Should this actually be called `times_next`? Given it appears to be *all* the time values, shifted by one step forward.
@seballgeyer

Comment (seballgeyer) [reply gnssanalysis/gn_io/sp3.py:490]: More a look into this function, obscurer it looks... 
it seems that the 20 lines functions is doing something that numpy can do by itself in 2 or 3 function calls. 

I would suggest to merge this PR and eventually rewrite this function later.

**Review by seballgeyer — APPROVED** [review]
More a look into this function, obscurer it looks...
it seems that the 20 lines functions is doing something that numpy can do by itself in 2 or 3 function calls.

I would suggest to merge this PR and eventually rewrite this function later. The objectives of renaming some variable to improve understanding has been achieved.

**Pull request merged**